### DISCUSSION TITLE
Calculations between tables

### DISCUSSION THREAD

**Discussion by jordie324** [Budibase/budibase#822] [Help] [answered]
Hi,

Is it possible to perform calculations between tables? For example, subtract the sum of Table A Column A from the sum of Table B Column B and have the difference show in Table C Column C?

Thanks!

**Comment by mjashanks** [comment]
Hi Jordan.

This is currently not possible.

It would be great to know a bit more about the use case though.

What do you plan to do with the "Difference in Table C Column C" ? What is the purpose of doing the calculation? Is it simply to show it in your app somewhere ?

If so, I would think that the way we could (in future) achieve this, is to be able to enter some formulas into the frontend...

For example, in a Heading component, we could set the text to something like `{{ TableA.Sum - TableB.Sum }}`. Again, this is not possble now, but I'm just trying to find the best possible way to achieve your goal.

Thanks for the feedback!

**Comment by jordie324** [comment, marked as answer]
Hi Michael,

Thanks for the reply!

I'm basically wanting to create a web app that is a combination of something like https://www.everydollar.com/ and the attached spreadsheet. I think front-end calculations would be very helpful! I do think I would need to store the calculated values in tables too though. Here are some specifics of what I'm trying to achieve:

- **Budgets**
I'd like users to be able to set budgets (income, expense groups, and expense items) and record income received and money spent. I'd like the app subtract income received and money spent from the budgeted amounts. I'd then like users to be able to create the next month's budget based on historical averages.
- **Debt**
I'd like users to be able to enter their various debts and either update the amount month-to-month or (possibly) auto-adjust the amount based on an interest rate and the amount paid down.
- **Savings**
I'd like users to be able to enter their various savings accounts and either update the amount month-to-month or (possibly) auto-adjust the amount based on an interest rate and the amount added.
- **Assets**
I'd like users to be able to enter their various assets and either update the value annually or (possibly) auto-adjust the amount based on depreciation and/or value minus amount owed.
- **Investments**
I'd like users to be able to enter their investment accounts and either update the value monthly/quarterly/annually or (possibly) auto-adjust the amount based on interest rate and amount added.
- **Net Worth**
I'd like to be able to add savings, assets, and investments together and subtract debt from the total to report net worth.
- **Year-Over-Year Reporting**
I'd like to show year-over-year trends, sliced and diced in multiple ways.
- **All Time Reporting**
I'd like to show trends for all data entered, sliced and diced in multiple ways.
- **Monthly Averages Reporting**
I'd like to show monthly averages for income and repeated expenses, sliced and diced in multiple ways.
- **Monthly & Yearly Projections**
Once enough data is entered, I'd like to be able to project income and expense amounts based on historical percentage changes.

Thank you!

[Finances-SCRUB.xlsx](https://github.com/Budibase/budibase/files/5495972/Finances-SCRUB.xlsx)

**Reply by mjashanks** [reply]
Amazing information, thanks so much Jordan.

I think the big thing that would top you creating something like this is formula columns, in tables. It's something that we have thought about for a while.

Whilst it is not on our immediate radar, I think we will have to tackle it at some point. It won't be in the next few months though :/

We are just about to release lovely charts though, which may be of interesting to your reporting... https://github.com/Budibase/budibase/pull/817

Date based views is another feature that we would need to tackle, to be able to complete some of your figures, which we currently don't support.

So, I don't think we are ready for your use case just yet, but thanks so much for sharing your use case.

**Comment by jordie324** [comment]
No problem, thank you for all your hard work! It's a great tool!
